lib/nrf_modem_lib/nrf91_connectivity/nrf91_connectivity.h

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF91_CONNECTIVITY_H__
+#define NRF91_CONNECTIVITY_H__
+
+#include <zephyr/types.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+
+/**
+ * @defgroup nrf91_connectivity nRF91 connectivity layer
+ * @{
+ * @brief Library that provides connectivity for nRF91 based boards.
+ *	  The library is intended to be integrated with and bound to a Zephyr network interface.
+ *	  It can be managed via Zephyr's network management API and the
+ *	  Connection Manager Connectivity API using calls such as net_if_down(), net_if_up(), and
+ *	  conn_mgr_if_connect().
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Options used to specify desired behavior when the network interface is brough down by
+ *	   calling net_if_down().
+ */
+enum nrf91_connectivity_if_down_options {
+	/** Shutdown modem - GNSS functionality will not be preserved. */
+	NRF91_CONNECTIVITY_IF_DOWN_MODEM_SHUTDOWN = 1,
+
+	/** Disconnect from LTE - GNSS functionality will be preserved. */
+	NRF91_CONNECTIVITY_IF_DOWN_LTE_DISCONNECT
+};
+
+/** @brief Option types used to specify library behavior. */
+enum nrf91_connectivity_options {
+	/** Sets desired behavior when nrf91_connectivity_disable() / net_if_down() is called. */
+	NRF91_CONNECTIVITY_IF_DOWN = 1,
+};
+
+/** @brief Macro used to define a private Connection Manager connectivity context type.
+ *         Required but not implemented.
+ */
+#define NRF91_CONNECTIVITY_CTX_TYPE void *
+
+/** @brief Initialize the nRF91 connectivity layer.
+ *
+ *  @details Bind to the network interface via the passed in binding structure.
+ *	     Setup appropriate internal handlers and mark the network interface as dormant.
+ *	     Set default options for connection timeout and persistency.
+ *
```

Comment:
I wonder if a note on these functions simply stating that they are part of the conn_mgr_conn_api struct functions would be a good idea

Not super necessary though if there isn't a succinct way to state it

Reply:
Good suggestion, added a comment in the general description of the library:

```
/**
 * @defgroup nrf91_connectivity nRF91 connectivity layer
 * @{
 * @brief Library that provides connectivity for nRF91 based boards.
 *	  The library is intended to be integrated with and bound to a Zephyr network interface via
 *	  the Connection Manager binding structure @ref conn_mgr_conn_api.
 *	  It can be managed via Zephyr's network management API and the
 *	  Connection Manager Connectivity API using calls such as net_if_down(), net_if_up(), and
 *	  conn_mgr_if_connect().
 */
```

Do you think this is sufficient?

Reply:
I think that is sufficient, aye!